### DISCUSSION TITLE
Stats API - period=realtime

### DISCUSSION THREAD

**Discussion by RobertJoonas** [plausible/analytics#1744] [Anything Else]
I would like to get a breakdown of my top pages in real time. This is currently impossible, as the shortest time period is one day and the `realtime` endpoint only gives the number of realtime visitors.

Would be nice if I could set the `period` parameter value to `realtime`. Then the API could interpret that as the last 5 minutes, and it would feel pretty much like real time.

(feedback via email)

**Comment by joshistoast** [comment]
In addition, I think adding the `filters` parameter to the `realtime` endpoint would be huge for us. Trying to add current visitors on a product page but can only return realtime views for the entire site.

**Comment by strefethen** [comment]
Plausible has a non-public API for this which it uses itself. A bit frustrating that it's not part of the supported APIs. The request is:

```https://<your_plausible_domain>/api/stats/ca-17.com/pages?period=realtime&date=2023-11-17&filters=%7B%7D&with_imported=true&limit=100&page=1&detailed=false```

It's not ideal because you'll will need to pass your dashboard cookie header. Btw, the "detailed" parameter seems to have no effect.

The result looks like this:

```json
[
  {
    "bounce_rate": null,
    "name": "/map",
    "pageviews": 10,
    "time_on_page": 32,
    "visitors": 7
  }
]
```

**Comment by antdickens** [comment]
Bump.

This works (using a Cookie header instead of API key). But why? Why isn't this part of the documented API?
